swatch-contracts/src/main/java/com/redhat/swatch/contract/repository/ContractRepository.java

```diff
@@ -20,97 +20,16 @@
  */
 package com.redhat.swatch.contract.repository;
 
-import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
 import java.util.List;
-import java.util.Map;
-import java.util.Optional;
 import java.util.UUID;
-import java.util.stream.Collectors;
 import javax.enterprise.context.ApplicationScoped;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 @ApplicationScoped
-public class ContractRepository implements PanacheRepositoryBase<ContractEntity, UUID> {
-
-  public List<ContractEntity> getContracts(Map<String, Object> parameters) {
-    if (parameters == null) {
-      return listAll();
-    }
-
-    Map<String, Object> nonNullParams =
-        parameters.entrySet().stream()
-            .filter(entry -> entry.getValue() != null)
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-
-    if (nonNullParams.isEmpty()) {
-      return listAll();
-    }
-
-    var propertiesRequiringJoin = List.of("metricId");
-    var isJoinTableNeeded = nonNullParams.containsKey("metricId");
-
-    String query =
-        nonNullParams.keySet().stream()
-            .map(
-                key -> {
-                  if (isJoinTableNeeded) {
-                    if (propertiesRequiringJoin.contains(key)) {
-                      return "m." + key + "=:" + key;
-                    } else {
-                      return "c." + key + "=:" + key;
-                    }
-                  }
-
-                  return key + "=:" + key;
-                })
-            .collect(Collectors.joining(" and "));
-
-    if (isJoinTableNeeded) {
-      var metricTableJoin =
-          "select c from "
-              + ContractEntity.class.getName()
-              + " c inner join "
-              + ContractMetricEntity.class.getName()
-              + " m on c.uuid = m.contractUuid where ";
-      query = metricTableJoin + query;
-    }
-
-    log.info("Dynamically generated query: {}", query);
-
-    return find(query, nonNullParams).list();
-  }
-
-  public Optional<ContractEntity> getContract(
```

Comment:
Before you remove this method could we make sure we have equivalent specification method?